### PR TITLE
drivers: sensor: Move to vendor subdirectory

### PR DESCRIPTION
**Pull request by MaureenHelm**
Organizes sensor drivers by vendor to distribute maintainership
responsibilities.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

### CODE REVIEW

**Comment by teburd** [comment]
very nice

**Comment by avisconti** [comment]
Not a bad idea and I personally like it.

File: drivers/sensor/CMakeLists.txt [hunk 1]
```diff
@@ -1,179 +1,73 @@
...
-add_subdirectory_ifdef(CONFIG_TMD2620 tmd2620)
-add_subdirectory_ifdef(CONFIG_TMP007 tmp007)
-add_subdirectory_ifdef(CONFIG_TMP108 tmp108)
-add_subdirectory_ifdef(CONFIG_TMP112 tmp112)
-add_subdirectory_ifdef(CONFIG_TMP116 tmp116)
-add_subdirectory_ifdef(CONFIG_TSL2540 tsl2540)
-add_subdirectory_ifdef(CONFIG_TSL2561 tsl2561)
-add_subdirectory_ifdef(CONFIG_TSL2591 tsl2591)
-add_subdirectory_ifdef(CONFIG_VCMP_IT8XXX2 ite_vcmp_it8xxx2)
 add_subdirectory_ifdef(CONFIG_VCNL4040 vcnl4040)
```
Comment (teburd) [inline drivers/sensor/CMakeLists.txt:67]: these three are vishay and could probably be moved as well correct?

File: drivers/sensor/Kconfig [hunk 2]
```diff
@@ -81,179 +81,74 @@ config SENSOR_INFO
...
-source "drivers/sensor/nuvoton_tach_npcx/Kconfig"
-source "drivers/sensor/nxp_kinetis_temp/Kconfig"
-source "drivers/sensor/nxp_tempmon/Kconfig"
-source "drivers/sensor/opt3001/Kconfig"
 source "drivers/sensor/pcnt_esp32/Kconfig"
 source "drivers/sensor/pms7003/Kconfig"
-source "drivers/sensor/qdec_mcux/Kconfig"
-source "drivers/sensor/qdec_nxp_s32/Kconfig"
-source "drivers/sensor/qdec_nrfx/Kconfig"
 source "drivers/sensor/qdec_sam/Kconfig"
```
Comment (teburd) [inline drivers/sensor/Kconfig:141]: microchip

File: drivers/sensor/CMakeLists.txt [hunk 3]
```diff
@@ -1,179 +1,73 @@
...
-add_subdirectory_ifdef(CONFIG_NPM1300_CHARGER npm1300_charger)
 add_subdirectory_ifdef(CONFIG_NTC_THERMISTOR ntc_thermistor)
-add_subdirectory_ifdef(CONFIG_NXP_TEMPMON nxp_tempmon)
-add_subdirectory_ifdef(CONFIG_OPT3001 opt3001)
 add_subdirectory_ifdef(CONFIG_PCNT_ESP32 pcnt_esp32)
 add_subdirectory_ifdef(CONFIG_PMS7003 pms7003)
-add_subdirectory_ifdef(CONFIG_QDEC_MCUX qdec_mcux)
-add_subdirectory_ifdef(CONFIG_QDEC_NXP_S32 qdec_nxp_s32)
-add_subdirectory_ifdef(CONFIG_QDEC_NRFX qdec_nrfx)
 add_subdirectory_ifdef(CONFIG_QDEC_SAM qdec_sam)
```
Comment (teburd) [inline drivers/sensor/CMakeLists.txt:60]: microchip

**Review by teburd — DISMISSED** [review]
Minor nits of some stuff that can be moved later or now

**Review by avisconti — APPROVED** [review]
LGTM
Reviewed the commit for ST part more carefully than the others

**Review by avisconti — CHANGES_REQUESTED** [review]
Sorry, I just noticed that two recent drivers are missing in ST subdir:

- iis328dq
- lis2dux12

Rest is fine. Thx!

**Review by teburd — APPROVED** [review]

**Review by avisconti — APPROVED** [review]

**Pull request merged**